### DISCUSSION TITLE
Splitting updates into two distinct groups

### DISCUSSION THREAD

**Discussion by ChristianIvicevic** [renovatebot/renovate#31140] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

In my projects I have almost always used a single package rule that batches all patch and minor updates into a single PR to reduce noise. Now that I'm experimenting with nx I need to handle it separately and rely on manually updating it since it sometimes requires additional migration steps that renovate doesn't do by itself. As such I had a separate nx group with `enabled: false` but decided to want a separate PR for it regardless just to be notified that updates do exist. However the config shown below does not separate updates into two PRs as expected and just adds nx dependencies into my batch group. Is there a way to nicely separate nx it out of my batch group?

```json5
{
  $schema: 'https://docs.renovatebot.com/renovate-schema.json',
  extends: [
    'config:best-practices',
    'schedule:weekly',
    ':disableRateLimiting',
    ':semanticCommitsDisabled',
    ':rebaseStalePrs',
    ':pinDependencies',
    ':pinDevDependencies'
  ],
  packageRules: [
    // Split nx updates into separate branch. (Doesn't work as expected.)
    {
      groupName: 'all nx dependencies',
      matchPackageNames: ['/@nx/*/', '/nx/'],
      groupSlug: 'nx-monorepo'
    },
    // Bunch up all non-major dependencies into a single PR. In the common case where the upgrades apply cleanly, this
    // causes less noise and is resolved faster than starting a bunch of upgrades in parallel for what may turn out to
    // be a suite of related packages all released at once.
    {
      groupName: 'all non-major dependencies',
      matchUpdateTypes: ['patch', 'minor'],
      groupSlug: 'all-minor-patch'
    }
  ]
}
```

**Comment by rarkins** [comment, marked as answer]
Reverse the order of your rules

**Reply by ChristianIvicevic** [reply]
Oh I wasn't aware of precedence being based on order of rules, but it makes sense now that I think of it. Thank you for pointing that out.
Would the fixed config be equivalent to extending `group:allNonMajor` followed by `group:nrwlMonorepo` to declutter manual rules?

**Reply by rarkins** [reply]
Yes, that too
